src/base/ApplicationCertificate.ts

```diff
@@ -39,47 +39,85 @@ export class ApplicationCertificate extends Resource {
       throw new Error('You need to pass either a zone id or a zone domain');
     }
 
-    const certificate = new AcmCertificate(this, `certificate`, {
-      domainName: config.domain,
+    const certificate = ApplicationCertificate.generateAcmCertificate(
+      this,
+      config.domain,
+      config.tags
+    );
+
+    const certificateRecord = ApplicationCertificate.generateRoute53Record(
+      this,
+      config.zoneId,
+      certificate
+    );
+
+    ApplicationCertificate.generateAcmCertificateValidation(
+      this,
+      certificate,
+      certificateRecord
+    );
+
+    this.arn = certificate.arn;
+  }
+
+  static generateAcmCertificate(
+    resource: Resource,
+    domain: string,
+    tags?: { [key: string]: string }
+  ): AcmCertificate {
+    return new AcmCertificate(resource, `certificate`, {
+      domainName: domain,
       validationMethod: 'DNS',
-      tags: config.tags,
+      tags: tags,
       lifecycle: {
         createBeforeDestroy: true,
       },
     });
+  }
 
-    const certificateRecord = new Route53Record(this, `certificate_record`, {
-      name: certificate.domainValidationOptions('0').resourceRecordName,
-      type: certificate.domainValidationOptions('0').resourceRecordType,
-      zoneId: config.zoneId,
+  static generateRoute53Record(
+    resource: Resource,
+    zoneId: string,
+    cert: AcmCertificate
+  ): Route53Record {
+    const record = new Route53Record(resource, `certificate_record`, {
+      name: cert.domainValidationOptions('0').resourceRecordName,
+      type: cert.domainValidationOptions('0').resourceRecordType,
+      zoneId,
       records: [],
       ttl: 60,
-      dependsOn: [certificate],
+      dependsOn: [cert],
```

Comment:
do we  want to include `alias` as a field? 

out of scope for this PR - question: if we want to configure health check for route53 or change routing rules, do they have to be instantiated in the base module?

Reply:
@sri-kirsh these are good questions that i don't immediately have an answer for. maybe good entries into the "what's puzzling us" section in the retro?